src/config/config_type.rs

```diff
@@ -30,17 +30,42 @@ pub enum ConfigType<'a> {
 }
 
 impl ConfigType<'_> {
-    /// Deserializes the configuration to `T` based on the input type. Errors if
-    /// the data produces an invalid config.
-    pub fn deserialize<T, P>(self, filter_name: &str) -> Result<T, Error>
+    /// Deserializes takes two type arguments `Static` and `Dynamic` representing
+    /// the types of a static and dynamic configuration respectively.
+    ///
+    /// If the configuration input is a [ConfigType::Static], then it is directly
+    /// marshalled into the returned `Static` instance.
```

Comment:
Very minor nit: The "Marshall" terminology isn't what you'd use in Rust. You'd normally just say "serialisation/deserialisation", I guess it's because in Rust they're not really objects in the same way they are in Go or Python. I don't think it's too important though.